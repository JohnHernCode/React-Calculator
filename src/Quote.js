import React, { useState, useEffect } from 'react';

function Quote() {
  const [quote, setQuote] = useState();
  const [author, setAuthor] = useState();
  // http://api.quotable.io/random

  useEffect(() => () => {
    fetch('https://api.quotable.io/random')
      .then((res) => res.json())
      .then(
        (quote) => {
          setQuote(quote.content);
          setAuthor(quote.author);
        },

      );
  }, []);

  return (
    <div className="quote">
      <h1>Quote of the day</h1>
      <p className="quote-text">
        {quote}
      </p>
      <p>{author}</p>
    </div>
  );
}

export default Quote;
