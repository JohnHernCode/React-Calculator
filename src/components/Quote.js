import React from 'react';
import styles from '../styles/Quote.module.css';
import img from '../images/sign.jpg';

const Quote = () => (
  <div className={styles.quote}>
    <h1 className={styles.head}>Quote of the day</h1>
    <p>
      Mathematics is not about numbers, equations, computations, or algorithms:
      it is about understanding.
    </p>
    <p>
      â€“William Paul ThurstonHome
    </p>
    <img src={img} alt="sign" />
  </div>
);

export default Quote;
