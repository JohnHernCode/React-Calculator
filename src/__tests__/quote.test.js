import React from 'react';
import renderer from 'react-test-renderer';
import Quote from '../components/Quote';

describe('Quote Component', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<Quote />);

    expect(tree).toMatchSnapshot();
  });
});
