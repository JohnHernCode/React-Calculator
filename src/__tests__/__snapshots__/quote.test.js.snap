// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote Component matches snapshot 1`] = `
<div
  className="quote"
>
  <h1
    className="head"
  >
    Quote of the day
  </h1>
  <p>
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding.
  </p>
  <p>
    â€“William Paul ThurstonHome
  </p>
  <img
    alt="sign"
    src="sign.jpg"
  />
</div>
`;
